server/src/main/java/io/druid/server/namespace/cache/LookupCoordinatorManager.java

```diff
@@ -0,0 +1,571 @@
+  {
+    final Map<String, Map<String, Map<String, Object>>> prior = getKnownLookups();
+    if (prior == null) {
+      LOG.warn("Requested delete lookup [%s]/[%s]. But no lookups exist!", tier, lookup);
+      return false;
+    }
+    final Map<String, Map<String, Map<String, Object>>> update = new HashMap<>(prior);
+    final Map<String, Map<String, Object>> tierMap = update.get(tier);
+    if (tierMap == null) {
+      LOG.warn("Requested delete of lookup [%s]/[%s] but tier does not exist!", tier, lookup);
+      return false;
+    }
+
+    if (!tierMap.containsKey(lookup)) {
+      LOG.warn("Requested delete of lookup [%s]/[%s] but lookup does not exist!", tier, lookup);
+      return false;
+    }
+
+    final Map<String, Map<String, Object>> newTierMap = new HashMap<>(tierMap);
+    newTierMap.remove(lookup);
+    update.put(tier, newTierMap);
+    configManager.set(LOOKUP_CONFIG_KEY, update, auditInfo);
+    return true;
+  }
+
+  /**
+   * Try to find a lookupName spec for the specified lookupName.
+   *
+   * @param lookupName The lookupName to look for
+   *
+   * @return The lookupName spec if found or null if not found or if no namespaces at all are found
+   */
+  public
+  @Nullable
+  Map<String, Object> getLookup(@NotNull final String tier, @NotNull final String lookupName)
+  {
+    final Map<String, Map<String, Map<String, Object>>> prior = getKnownLookups();
+    if (prior == null) {
+      LOG.warn("Requested tier [%s] lookupName [%s]. But no namespaces exist!", tier, lookupName);
+      return null;
+    }
+    final Map<String, Map<String, Object>> tierLookups = prior.get(tier);
+    if (tierLookups == null) {
+      LOG.warn("Tier [%s] does not exist", tier);
+      return null;
+    }
+    return tierLookups.get(lookupName);
+  }
+
+
+  @LifecycleStart
+  public void start()
+  {
+    synchronized (startStopSync) {
+      if (started) {
+        return;
+      }
+      if (executorService.isShutdown()) {
+        throw new ISE("Cannot restart after stop!");
+      }
+      lookupMapConfigRef = configManager.watch(
+          LOOKUP_CONFIG_KEY,
+          new TypeReference<Map<String, Map<String, Map<String, Object>>>>()
+          {
+          },
+          null
+      );
+      executorService.scheduleWithFixedDelay(
+          new Runnable()
+          {
+            @Override
+            public void run()
+            {
+              final Map<String, Map<String, Map<String, Object>>> allLookupTiers = lookupMapConfigRef.get();
+              // Sanity check for if we are shutting down
+              if (Thread.currentThread().isInterrupted()) {
+                LOG.info("Not updating namespace lookups because process was interrupted");
+                return;
+              }
+              if (!started) {
+                LOG.info("Not started. Returning");
+                return;
+              }
+              for (final String tier : allLookupTiers.keySet()) {
```

Comment:
again this will not see all the announced hosts thought, you will only see what the user has added as lookups from the coordinator side